### PR TITLE
[PKD-29]: integrate serve real data

### LINKED ISSUE #29: Serve real data

### PR DESCRIPTION
**Pull request by eidan66**

### CODE REVIEW

File: server/app.ts [hunk 1]
```diff
@@ -1,13 +1,65 @@
 import * as http from 'http';
-import MOCK_DATA from './data/mock-data.js'
+import * as querystring from 'querystring';
+import * as url from 'url';
+import { Pokemon, PokemonClient } from 'pokenode-ts';
 
-const server = http.createServer((req, res) => {
-    const url = req.url;
+import COLORS from './colors.js';
+import { APIPageResponse, APIPokemon, APIPokemonType } from './responseTypes.js';
```
Comment (eidan66) [inline server/app.ts:7]:
```suggestion
import COLORS from './colors';
import { APIPageResponse, APIPokemon, APIPokemonType } from './responseTypes';
```

File: server/colors.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    grass: '#7AC74C',
+    ground: '#E2BF65',
+    ice: '#96D9D6',
+    normal: '#A8A77A',
+    poison: '#A33EA1',
+    psychic: '#F95587',
+    rock: '#B6A136',
+    steel: '#B7B7CE',
+    water: '#6390F0',
+  } as  { [key: string]: string };
```
Comment (eidan66) [inline server/colors.ts]: The `key:string` is the pokemon type, can you use it instead "random" string?

File: server/responseTypes.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    Ice = 'Ice',
+    Normal = 'Normal',
+    Poison = 'Poison',
+    Psychic = 'Psychic',
+    Rock = 'Rock',
+    Steel = 'Steel',
+    Water = 'Water',
+
+    Unknown = 'Unknown'
+  }
```
Comment (eidan66) [inline server/responseTypes.ts]: If you use `APIPokemonType` inside `APIPokemon` please move `APIPokemonType` before `APIPokemon`.

```suggestion
export enum APIPokemonType {
    Bug = 'Bug',
    Dark = 'Dark',
    Dragon = 'Dragon',
    Electric = 'Electric',
    Fairy = 'Fairy',
    Fighting = 'Fighting',
    Fire = 'Fire',
    Flying = 'Flying',
    Ghost = 'Ghost',
    Grass = 'Grass',
    Ground = 'Ground',
    Ice = 'Ice',
    Normal = 'Normal',
    Poison = 'Poison',
    Psychic = 'Psychic',
    Rock = 'Rock',
    Steel = 'Steel',
    Water = 'Water',

    Unknown = 'Unknown'
  }
  
  export interface APIPokemon {
    pokemonName: string, 
    pokemonNumber: string, 
    pokemonTypes: APIPokemonType[], 
    boxBg: string, 
    pokemonSvg: string
}
```

File: yarn.lock [hunk 4]
```diff
@@ -0,0 +1,89 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Comment (eidan66) [inline yarn.lock]: Do we need this yarn.lock at the root of the repo?

File: server/colors.ts (hunk 2 above)
Comment (almogko) [reply server/colors.ts]: i can't use PokemonType as the key type

Comment (eidan66) [reply server/colors.ts]:
```suggestion
  } as  { [key in APIPokemonType]: string };
```

Something like this

File: server/app.ts [hunk 5]
```diff
@@ -16,4 +68,37 @@ const server = http.createServer((req, res) => {
...
+    return `#00${num}`
+}
+
+function pokeAPIPokemonToAPIPokemon(pokeApiPokemon: Pokemon): APIPokemon {
+    return {
+        pokemonName: pokeApiPokemon.name,
+        pokemonNumber: pad(pokeApiPokemon.id),
+        pokemonTypes: pokeApiPokemon.types.map(({type:{name}})=> mapToAPIPokemonType(name)),
+        boxBg: COLORS[mapToAPIPokemonType(pokeApiPokemon.types[0].type.name)],
+        pokemonSvg: `https://github.com/eidan66/pokemon-api-sprites/blob/master/sprites/pokemon/other/showdown/${pokeApiPokemon.id}.gif?raw=true`
```
Comment (eidan66) [inline server/app.ts]: Can you return also `pokemonId` please?
```suggestion
        pokemonName: pokeApiPokemon.name,
        pokemonId:pokeApiPokemon.id),
        pokemonNumber: pad(pokeApiPokemon.id),
        pokemonTypes: pokeApiPokemon.types.map(({type:{name}})=> mapToAPIPokemonType(name)),
        boxBg: COLORS[mapToAPIPokemonType(pokeApiPokemon.types[0].type.name)],
        pokemonSvg: `https://github.com/eidan66/pokemon-api-sprites/blob/master/sprites/pokemon/other/showdown/${pokeApiPokemon.id}.gif?raw=true`
```

File: server/responseTypes.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
```
Comment (eidan66) [inline server/responseTypes.ts:30]: Remove extra line

**Pull request merged**